### DISCUSSION TITLE
Using Interval with "serverless" services

### DISCUSSION THREAD

**Discussion by jpwilliams** [interval/interval-node#2] [Q&A] [answered]
Hi folks! First off, fantastic product - really easy to understand what's going on and it's so, so powerful. Stellar job.

I wondered what the thoughts were on using Interval within "serverless" services as opposed to those with a constant uptime.

I can see from the [Security](https://interval.com/docs/security) page that Interval establishes "_a persistent bi-directional message passing link between [it and my app]_"; I'm guessing from the word "persistent" here that it requires that my app remains active and connected at all times.

This would make sense, as I'm guessing Interval has to keep track of which hosts are running and contactable to be able to run actions. All the same, I'd love to know if there's something I'm missing in the docs or if Interval has serverless support on the horizon.

**Comment by alexarena** [comment, marked as answer]
Hey Jack, really glad you like it! 

You're right about how our architecture works at the moment. Great timing for your question though: we've actually got serverless support working internally now and are getting close to a beta release. 

If you let me know your account email (either leave it in this thread or feel free to message me privately, I'm alex@interval.com) we can make sure you get access to the beta as soon as its ready.

Out of curiosity, what serverless cloud provider would you want to use? Each one is a little different, so we're trying to gauge interest.

**Reply by jpwilliams** [reply]
Hi @alexarena!

My particular need was to lazily run some code in a serverless function hosted in Vercel - we've found we have quite a few UIs hosted on there that inevitably grow small APIs of their own and Interval would be great to wrangle them in to submission.

The majority of the "services" we want to hook in to tend to be BFFs for projects that started as simple UIs; things that aren't naturally very accessible and/or weren't created with connectivity in mind. We do have Lambdas about, but on reflection they're a lot more purposeful (ETL, auth, etc) so I can't think of an immediate need for that in my particular case.

I'll send you an email to send across my mail address. 😃
